### PR TITLE
chore: frm api changes

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Changes made in FRM display fields and api 

Current
image
image

New
image

## Motivation and Context
align frontend with backend api changes

## How did you test it?
- New UI is as expected
- API are giving expected responses
- New and update flow for frm connectors tested for new and old users
- Checked and tested with backend team
- Test Payment is working with FRM enabled and disabled

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by sagarnaikjuspay — APPROVED** [review]

File: src/screens/FraudAndRisk/FRMTypes.res [hunk 1]
```diff
@@ -1,6 +1,6 @@
 type filterType = Connector | FRMPlayer | ThreedsAuthenticator
 
-type frmPaymentMethodsSectionType = FlowType | ActionType
+type frmPaymentMethodsSectionType = FlowType
```
Comment (sagarnaikjuspay) [inline src/screens/FraudAndRisk/FRMTypes.res:3]: do we need a type define when there is only one type available

File: src/screens/FraudAndRisk/FRMUtils.res [hunk 2]
```diff
@@ -134,27 +134,17 @@ let createAllOptions = connectorsConfig => {
...
-          flow: "pre",
-          action: "cancel_txn",
-        }
-      })
-
-    {
+    let b: ConnectorTypes.frm_payment_method = {
       payment_method: paymentMethodName,
-      payment_method_types: paymentMethodTypesArr,
+      flow: "pre",
```
Comment (sagarnaikjuspay) [inline src/screens/FraudAndRisk/FRMUtils.res:145]: this value can be a type defined and use mapper to get string value

File: src/screens/FraudAndRisk/FRMSummary.res [hunk 3]
```diff
@@ -31,20 +27,30 @@ module ConfigInfo = {
...
+              ->Option.getOr([])
               ->Array.mapWithIndex(
                 (paymentMethodType, index) => {
-                  <InfoField
-                    key={index->Int.toString}
-                    label={paymentMethodType.payment_method_type}
-                    flowTypeValue={paymentMethodType.flow}
-                    actionTypeValue={paymentMethodType.action}
-                  />
+                  if index == 0 {
```
Comment (PritishBudhiraja) [inline src/screens/FraudAndRisk/FRMSummary.res:41]: Renderif

**Review by PritishBudhiraja — APPROVED** [review]

File: src/screens/Connectors/ConnectorTypes.res [hunk 4]
```diff
@@ -226,7 +226,8 @@ type frm_payment_method_type = {
 
 type frm_payment_method = {
   payment_method: string,
-  payment_method_types: array<frm_payment_method_type>,
+  payment_method_types?: array<frm_payment_method_type>,
+  mutable flow: string,
```
[the remark this answers is not in the document]
Comment (gitanjli525) [reply src/screens/Connectors/ConnectorTypes.res]: yeah thanks, not required

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by sagarnaikjuspay — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Pull request merged**